tests/bmtestspolar/symfixedrankpolarfactory.m

```diff
@@ -0,0 +1,215 @@
+function M = symfixedrankpolarfactory(m, k)
+% Manifold of m-by-m symmetric positive definite matrices of rank k with polar quotient geometry.
+%
+% function M = symfixedrankpolarfactory(m, k)
+%
+% The first-order geometry follows the balanced quotient geometry described
+% in the paper,
+% "Linear regression under fixed-rank constraints: a Riemannian approach",
+% G. Meyer, B. Bonnabel and R. Sepulchre, ICML 2011.
+%
+% Paper link: http://www.icml-2011.org/papers/350_icmlpaper.pdf.
+%
+% The second-order geometry follows from the paper
+% "Fixed-rank matrix factorizations and Riemannian low-rank optimization",
+% B. Mishra, R. Meyer, B. Bonnabel and R. Sepulchre,
+% Computational Statistics, 29(3 - 4), pp. 591 - 621, 2014.
+%
+% A point X on the manifold is represented as a structure with two
+% fields: U and B. The matrix U (mxk) is orthonormal,
+% while the matrix B (kxk) is a symmetric positive definite full rank
+% matrix.
+%
+% Tangent vectors are represented as a structure with two fields: U and B.
+%
+%
+% For first-order geometry, please cite the Manopt paper as well as the research paper:
+%     @InProceedings{meyer2011linear,
+%       Title        = {Linear regression under fixed-rank constraints: a {R}iemannian approach},
+%       Author       = {Meyer, G. and Bonnabel, B. and Sepulchre, R.},
+%       Booktitle    = {{28th International Conference on Machine Learning}},
+%       Year         = {2011},
+%       Organization = {{ICML}}
+%     }
+% For second-order geometry, please cite the Manopt paper as well as the research paper:
+%     @Article{mishra2014fixedrank,
+%       Title   = {Fixed-rank matrix factorizations and {Riemannian} low-rank optimization},
+%       Author  = {Mishra, B. and Meyer, G. and Bonnabel, B. and Sepulchre, R.},
+%       Journal = {Computational Statistics},
+%       Year    = {2014},
+%       Number  = {3-4},
+%       Pages   = {591--621},
+%       Volume  = {29},
+%       Doi     = {10.1007/s00180-013-0464-z}
+%     }
+%
+%
+% See also symfixedrankYYfactory sympositivedefinitefactory
+    
+    % This file is part of Manopt: www.manopt.org.
+    % Original author: Bamdev Mishra, April 05, 2019.
+    % Contributors:
+    % Change log:
+    %
+    
+    M.name = @() sprintf('UBU'' quotient manifold of %dx%d matrices of rank %d', m, m, k);
+    
+    M.dim = @() m*k - k*(k-1)/2;
+    
+    % Choice of the metric on the orthnormal space is motivated by the symmetry present in the
+    % space. The metric on the positive definite space is its natural metric.
+    M.inner = @(X, eta, zeta) eta.U(:).'*zeta.U(:)  ...
+        + trace( (X.B\eta.B) * (X.B\zeta.B) );
+    
+    M.norm = @(X, eta) sqrt(M.inner(X, eta, eta));
```

Comment:
Current code will solve the linear system X.B\eta.B twice; it's not a big problem though; this is fine as is.